storm-core/src/jvm/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -0,0 +1,494 @@
+        String arch = System.getProperty("os.arch");
+        String archResourceRoot = resourceRoot + Utils.FILE_PATH_SEPARATOR + os + "-" + arch;
+        String ret = archResourceRoot + Utils.CLASS_PATH_SEPARATOR + resourceRoot + Utils.CLASS_PATH_SEPARATOR + conf.get(Config.JAVA_LIBRARY_PATH);
+        return ret;
+    }
+    
+    @SuppressWarnings("unchecked")
+    protected String getWorkerClassPath(String stormJar, Map<String, Object> topologyConf) {
+        List<String> topoClasspath = new ArrayList<>();
+        Object object = topologyConf.get(Config.TOPOLOGY_CLASSPATH);
+
+        // Will be populated only if STORM_TOPOLOGY_CLASSPATH_BEGINNING_ENABLED is set on Nimbus.
+        // Allowed for extreme debugging.
+        Object topologyClasspathFirst = topologyConf.get(Config.TOPOLOGY_CLASSPATH_BEGINNING);
+        List<String> firstClasspathList = new ArrayList<>();
+        if(topologyClasspathFirst instanceof List) {
+            firstClasspathList.addAll((List<String>)topologyClasspathFirst);
+        } else if (topologyClasspathFirst instanceof String) {
+            firstClasspathList.add((String) topologyClasspathFirst);
+        }
+        LOG.debug("Topology Classpath Prefix: {}", firstClasspathList);
+
+        if (object instanceof List) {
+            topoClasspath.addAll((List<String>) object);
+        } else if (object instanceof String) {
+            topoClasspath.add((String) object);
+        }
+        LOG.debug("Topology specific classpath is {}", object);
+
+        String classPath = Utils.addToClasspath(firstClasspathList, Arrays.asList(Utils.workerClasspath()));
+        String classAddPath = Utils.addToClasspath(classPath, Arrays.asList(stormJar));
+        return Utils.addToClasspath(classAddPath, topoClasspath);
+    }
+    
+    private static String substituteChildOptsInternal(String string,  String workerId, String stormId, int port, int memOnheap) {
+        if (StringUtils.isNotBlank(string)){
+            string = string.replace("%ID%", String.valueOf(port));
+            string = string.replace("%WORKER-ID%", workerId);
+            string = string.replace("%TOPOLOGY-ID%", stormId);
+            string = string.replace("%WORKER-PORT%", String.valueOf(port));
+            string = string.replace("%HEAP-MEM%", String.valueOf(memOnheap));
+        }
+        return string;
+    }
+    
+    protected List<String> substituteChildopts(Object value, String workerId, String stormId, int port, int memOnheap) {
+        List<String> rets = new ArrayList<>();
+        if (value instanceof String) {
+            String string = substituteChildOptsInternal((String) value,  workerId, stormId, port, memOnheap);
+            if (StringUtils.isNotBlank(string)){
+                String[] strings = string.split("\\s+");
+                rets.addAll(Arrays.asList(strings));
+            }
+        } else if (value instanceof List) {
+            @SuppressWarnings("unchecked")
```

Comment:
I put it here because it limits the scope.  That way if someone did something bad in another part of the same method it would still be a warning.